xbmc/addons/gui/GUIDialogAddonInfo.cpp

```diff
@@ -770,6 +769,15 @@ void CGUIDialogAddonInfo::BuildDependencyList()
       addonAvailable = nullptr;
     }
 
+    if (!addonInstalled)
+    {
+      if (showAllDependencies || !addonAvailable ||
+          addonAvailable->MainType() != ADDON_SCRIPT_MODULE)
+      {
+        m_showDepDialogOnInstall = true;
+      }
+    }
+
```

Comment:
A comment would make sense here. I'm not entirely sure why we should show deps on install if an addon is not available in repos?

Reply:
so the user will get a chance to see why the following installation fails. as the „not available“ dialog pops up right after clicking the install button. best option would be the installation is cancelled in this case.

Reply:
would be possible to disable the install button if a dep is unavailable. but then nobody has a clue why it‘s disabled unless they look under dependencies...

Reply:
Definitely needs a comment in the code!!!